### PR TITLE
Add JDBC v3.1.0 for Flink 1.16.x and Flink 1.17.x

### PR DESCRIPTION
**Pull request by dannycranmer**

### CODE REVIEW

File: docs/data/release_archive.yml [hunk 1]
```diff
@@ -428,6 +428,11 @@ release_archive:
       version: 1.0.0
       release_date: 2022-12-21
       filename: "opensearch"
+    - name: "Flink JDBC Connector"
+      connector: "jdbc"
+      version: 3.0.0
```
Comment (elphastori) [inline docs/data/release_archive.yml]: 3.1.0

**Review by elphastori — CHANGES_REQUESTED** [review]
Checked local build, archive version needs updating. The rest looks good.

File: docs/data/release_archive.yml [hunk 2]
```diff
@@ -428,6 +428,11 @@ release_archive:
...
+      release_date: 2023-04-13
```
Comment (libenchao) [inline docs/data/release_archive.yml]: Do we need to make sure that it's the same with the time we publish to Nexus?

**Review by MartijnVisser — APPROVED** [review]
LGTM

**Review by elphastori — APPROVED** [review]

File: docs/data/release_archive.yml (hunk 2 above)
Comment (leonardBang) [reply docs/data/release_archive.yml]: Add a FIXME note is more suggested

**Review by leonardBang — APPROVED** [review]

**Pull request merged**